pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/bindings/LTensorTNCudaBindings.hpp

```diff
@@ -114,68 +118,113 @@ void registerBackendClassSpecificBindings(PyClass &pyclass) {
     using np_arr_c = py::array_t<std::complex<ParamT>,
                                  py::array::c_style | py::array::forcecast>;
 
-    pyclass
-        .def(py::init<const std::size_t,
-                      const std::size_t>()) // num_qubits, max_bond_dim
-        .def(py::init<const std::size_t, const std::size_t,
-                      DevTag<int>>()) // num_qubits, max_bond_dim, dev-tag
-        .def(
-            "getState",
-            [](TensorNet &tensor_network, np_arr_c &state) {
-                py::buffer_info numpyArrayInfo = state.request();
-                auto *data_ptr =
-                    static_cast<std::complex<PrecisionT> *>(numpyArrayInfo.ptr);
-
-                tensor_network.getData(data_ptr, state.size());
-            },
-            "Copy StateVector data into a Numpy array.")
-        .def("applyControlledMatrix", &applyControlledMatrix<TensorNet>,
-             "Apply controlled operation")
-        .def(
-            "updateMPSSitesData",
-            [](TensorNet &tensor_network, std::vector<np_arr_c> &tensors) {
-                for (std::size_t idx = 0; idx < tensors.size(); idx++) {
-                    py::buffer_info numpyArrayInfo = tensors[idx].request();
+    if constexpr (std::is_same_v<TensorNet, MPSTNCuda<double>> ||
```

Comment:
Would it be more concise if we put the methods in the `TNCuda class` together and then expose MPS specified method in a separate method?

Reply:
Do you mean to make the binding for the base class `TNCuda` instead of child classes `MPS` and `ExactTN`? :thinking: